components/file_system/src/file.rs

```diff
@@ -231,6 +234,19 @@ impl Default for OpenOptions {
     }
 }
 
+#[cfg(target_os = "linux")]
+impl OpenOptionsExt for OpenOptions {
```

Comment:
Why reinvent the wheel?

Reply:
`file_system` is supposed to replace `std::fs`, therefore traits that work for `std::fs::OpenOptions` should also work for `file_system::OpenOptions`.

Reply:
What I mean is why replace `std::fs`? Why not partially replaced? Or provide patch functions?

Reply:
Partial replacement won't help this case, `OpenOptions` is tightly coupled with `File` type. Patch function like `fn wrap(std::File) -> file_system::File` is inconvenient compared to replacing. Such [trivial inconvenience](https://www.lesswrong.com/posts/reitXJgJXFzKpdKyd/beware-trivial-inconveniences) is enough to discourage other contributors to use our customized file.